src/ComposerRequireChecker/Cli/ResultsWriter/ResultsWriter.php

```diff
@@ -0,0 +1,13 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ComposerRequireChecker\Cli\ResultsWriter;
+
+interface ResultsWriter
+{
+    /**
+     * @param array<array-key, string[]> $unknownSymbols the unknown symbols found
```

Comment:
Is there something more precise than `string[]` here? :thinking: `list<string>`, or even `list<non-empty-string>` perhaps?

Reply:
`list<string>` is possible I guess. `non-empty-string` does not work because we only detect symbols source from project or extensions but not from transient dependencies yet.